lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -756,8 +809,18 @@ function Invoke-InstallHook {
         Write-JujuWarning "Failed to set power scheme."
     }
     Start-TimeResync
+    
+    $cfg = Get-JujuCharmConfig
+    if ($cfg['format-raw-devices']) {
+        Get-Disk | Where partitionstyle -eq 'raw' | `
+        Initialize-Disk -PartitionStyle GPT -PassThru | `
+        New-Partition -AssignDriveLetter -UseMaximumSize | `
+        Format-Volume -FileSystem NTFS -Confirm:$false
+    }
+    
+    $mpioReboot = Enable-MPIO
```

Comment:
I think it would make sense to move this call to ```Enable-MPIO```  in ```Install-Prerequisites```. Then we can simply set ```$prereqReboot``` accordingly and leave the reboot check as it was.